vcd/resource_vcd_edgegateway.go

```diff
@@ -145,12 +168,107 @@ func resourceVcdEdgeGatewayCreate(d *schema.ResourceData, meta interface{}) erro
 		log.Printf("[DEBUG] Error creating edge gateway: %#v", err)
 		return fmt.Errorf("error creating edge gateway: %#v", err)
 	}
+	// Edge gateway creation succeeded therefore we save related fields now to preserve Id.
+	// Edge gateway is already created even if further process fails
+	log.Printf("[TRACE] flushing edge gateway creation fields")
+	err = setEdgeGatewayValues(d, edge)
+	if err != nil {
+		return err
+	}
+
+	// Only perform general load balancer configuration if settings are set
+	if isEdgeGatewayLbConfigured(d) {
+		if !d.Get("advanced").(bool) {
+			return fmt.Errorf("load balancing cannot be used when advanced networking is disabled")
+		}
+
+		log.Printf("[TRACE] edge gateway load balancer configuration started")
+
+		err := updateLoadBalancer(d, edge)
+		if err != nil {
+			return fmt.Errorf("unable to update general load balancer settings: %s", err)
+		}
+
+		log.Printf("[TRACE] edge gateway load balancer configured")
+	}
 
 	d.SetId(edge.EdgeGateway.ID)
 	log.Printf("[TRACE] edge gateway created: %#v", edge.EdgeGateway.Name)
 	return resourceVcdEdgeGatewayRead(d, meta)
 }
 
+// Fetches information about an existing edge gateway for a data definition
+func resourceVcdEdgeGatewayRead(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[TRACE] edge gateway read initiated")
+
+	vcdClient := meta.(*VCDClient)
+
+	edgeGateway, err := vcdClient.GetEdgeGatewayFromResource(d, "name")
+	if err != nil {
+		d.SetId("")
+		return nil
+	}
+
+	if err := setEdgeGatewayValues(d, edgeGateway); err != nil {
+		return err
+	}
+
+	// Only read and set the statefile if the edge gateway is advanced
+	// and general lb settings are used
+	if isEdgeGatewayLbConfigured(d) && edgeGateway.HasAdvancedNetworking() {
+		if err := setLoadBalancerData(d, edgeGateway); err != nil {
+			return err
+		}
+	}
+
+	log.Printf("[TRACE] edge gateway read completed: %#v", edgeGateway.EdgeGateway)
+	return nil
+}
+
+// resourceVcdEdgeGatewayUpdate updates an edge gateway from a resource definition
+func resourceVcdEdgeGatewayUpdate(d *schema.ResourceData, meta interface{}) error {
```

Comment:
We have an issue about implementing Update for EGW. Please add a note saying that this Update is only dealing with LB items so far.

Reply:
Then maybe we need create Issue to tackle full edge gtw update?

Reply:
we have one https://github.com/vmware/go-vcloud-director/issues/202